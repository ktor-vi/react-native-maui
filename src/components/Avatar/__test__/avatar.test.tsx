import { render } from '@testing-library/react-native';
import {
  withReanimatedTimer
} from 'react-native-reanimated/src/reanimated2/jestUtils'
import {Text} from 'react-native';
import {Avatar} from '../index';

describe('Test:Avatar', () => {
  it('render correctly', () => {
    withReanimatedTimer(() => {
      const tree = render(
        <Avatar
          style={{ width: 60, height: 60, borderRadius: 30 }}
          url={'https://avatars.githubusercontent.com/u/16695567?s=400&u=fd8d6249fa408e1e606015a06868a99993171938&v=4'}
          placeholder={<Text>加载中</Text>}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('base', () => {
    withReanimatedTimer(() => {
      const { queryByText } = render(
        <Avatar
          style={{ width: 60, height: 60, borderRadius: 30 }}
          url={'https://avatars.githubusercontent.com/u/16695567?s=400&u=fd8d6249fa408e1e606015a06868a99993171938&v=4'}
          placeholder={<Text>加载中</Text>}
        />
      );
      const element = queryByText('加载中');
      expect(element).not.toBeNull();
    });
  });
});
