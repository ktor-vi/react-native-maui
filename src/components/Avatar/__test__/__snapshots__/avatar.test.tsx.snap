// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test:Avatar render correctly 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "animatedStyle": Object {
          "current": Object {},
        },
        "initial": Object {
          "updater": [Function],
          "value": Object {
            "opacity": 0,
          },
        },
        "viewDescriptors": Object {
          "add": [Function],
          "batchToRemove": Object {},
          "items": Array [],
          "rebuildsharableViewDescriptors": [Function],
          "remove": [Function],
          "sharableViewDescriptors": Object {
            "_animation": null,
            "_id": 3,
            "_listeners": Array [],
            "_setter": [Function],
            "_value": Array [],
          },
          "tags": Object {},
          "waitForInsertSync": false,
          "waitForRemoveSync": false,
        },
        "viewsRef": Object {
          "add": [Function],
          "items": Object {},
          "remove": [Function],
        },
      }
    }
  >
    <Image
      onLoad={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://avatars.githubusercontent.com/u/16695567?s=400&u=fd8d6249fa408e1e606015a06868a99993171938&v=4",
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": 60,
          "width": 60,
        }
      }
    />
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#eee",
        "borderRadius": 30,
        "bottom": 0,
        "height": 60,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 60,
      }
    }
  >
    <Text>
      加载中
    </Text>
  </View>
</View>
`;
